### PR TITLE
CVE-2016-1647 and CVE-2016-1671

### PR DESCRIPTION
**Pull request by avj2668**
Creating Pull Request

### CODE REVIEW

File: cves/CVE-2016-1647.yml [hunk 1]
```diff
@@ -32,41 +32,45 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  The navigation implementation in Google Chrome allows remote attackers to
+  cause a denial of service, allows unauthorized disclosure of information,
```
Comment (njb5081) [inline cves/CVE-2016-1647.yml]: Seems a little vague, considering adding that it is a use-after-free vulnerability and maybe use parts of the description given by the finder in the bug report here: https://bugs.chromium.org/p/chromium/issues/detail?id=590284

File: cves/CVE-2016-1647.yml [hunk 2]
```diff
@@ -97,20 +103,21 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    No evidence found
```
Comment (njb5081) [inline cves/CVE-2016-1647.yml]: Didn't see much here either, could mention that it looks like its an external user but that's about it

File: cves/CVE-2016-1647.yml [hunk 3]
```diff
@@ -197,4 +208,21 @@ mistakes:
...
+    This happens for example when only the url fragment changes.  In such case, renderer sets
+    FrameHostMsg_DidCommitProvisionalLoad_Params::was_within_same_page = true and the
+    browser mostly trusts that, even if the the navigation wasn't within page and the
+    process was swapped. If the renderer swaps the process and sets was_within_same_page = true,
+    the fullscreen_widget_routing_id_ will end up pointing to a wrong widget in the new process.
+    If the renderer now closes fullscreen, this wrong widget is deleted. It is deleted manually
+    in RenderWidgetHostImpl::Destroy(), because fullscreen widgets aren't owned by a smart pointer.
+    Normal widgets are owned by a scoped_ptr render_widget_host_ in RenderViewHostImpl though.
+    If such a widget is manually deleted by an attacker, the scoped_ptr in RVHI ends up holding a
+    dangling pointer.
```
Comment (njb5081) [inline cves/CVE-2016-1647.yml]: Very similar  to the bug report description, consider rewriting in your own voice.

File: cves/CVE-2016-1647.yml [hunk 4]
```diff
@@ -32,41 +32,45 @@ description_instructions: |
...
+  note: Doesn't check the condition if navigation within page
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes:
```
Comment (njb5081) [inline cves/CVE-2016-1647.yml]: I'll give this 2 upvotes.

File: cves/CVE-2016-1671.yml [hunk 5]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Google Chrome on Android mishandles / (slash) and \ (backslash) characters,
+  which allows attackers to conduct directory traversal attacks via using a file
+  containing URLs.
```
Comment (njb5081) [inline cves/CVE-2016-1671.yml]: Good concise description, just reword in your own voice to differentiate it from the CVE description.

File: cves/CVE-2016-1671.yml [hunk 6]
```diff
@@ -46,22 +49,24 @@ reviews:
...
 fixes_vcc_instructions: |
   Please put the commit hash in "commit" below (see my example in
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 30408ae67a9f6aea074b2883ba861613f52bd246
   :note: ''
-vccs: []
+vccs:
+- commit: 2cd905494ba700a8e2097d614ed39ad36d1519f2
+  note: The person does not check for specific conditions from the outset
```
Comment (njb5081) [inline cves/CVE-2016-1671.yml]: What specific conditions?

File: cves/CVE-2016-1671.yml [hunk 7]
```diff
@@ -46,22 +49,24 @@ reviews:
...
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes:
```
Comment (njb5081) [inline cves/CVE-2016-1671.yml]: This seems like something everyone should know but still ends up happening, I'll give this 4 upvotes.

File: cves/CVE-2016-1671.yml [hunk 8]
```diff
@@ -92,20 +99,22 @@ discovered:
...
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    Developer mmenke@chromium.org	spotted the bug and brought it to the notice
+    of the reviewers.
```
Comment (njb5081) [inline cves/CVE-2016-1671.yml]: It appears credit for the bug was given to Jann Horn on the Stable Channel Update.
https://chromereleases.googleblog.com/2016/05/stable-channel-update.html

File: cves/CVE-2016-1671.yml [hunk 9]
```diff
@@ -192,4 +211,10 @@ mistakes:
...
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    To begin with, it was a coding mistake where the developer did not check for
+    specific escape characters (i.e. slash and backslash). The fix was pretty
+    straightforward in the escape.cc file.
+
+    The vulenrability was introduced when some new logical checks were being
+    implemented. The developer mishandled / (slash) and \ (backslash) characters.
```
Comment (njb5081) [inline cves/CVE-2016-1671.yml]: Seems pretty self explanatory, you might be able to expand upon it a little bit by describing how they mishandled escape characters.

File: cves/CVE-2016-1647.yml [hunk 10]
```diff
@@ -32,41 +32,46 @@ description_instructions: |
...
 bounty_instructions: |
   If you came across any indications that a bounty was paid out for this
   vulnerability, fill it out here. Or correct it if the information already here
   was wrong. Otherwise, leave it blank.
 bounty:
-  date: 2016-03-24 13:55:00.000000000 -04:00
-  amount: 5500.0
-  references:
-  - http://chromereleases.googleblog.com/2016/03/stable-channel-update_24.html
+  amt: 5500.0
```
Comment (Nonordon) [inline cves/CVE-2016-1647.yml]: It looks like the actual reward may have been $10,500 according to comment 35 here: https://bugs.chromium.org/p/chromium/issues/detail?id=590284

File: cves/CVE-2016-1647.yml [hunk 11]
```diff
@@ -32,41 +32,46 @@ description_instructions: |
...
   Please put the commit hash in "commit" below (see my example in
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
-- :commit: e5787005a9004d7be289cc649c6ae4f3051996cd
-  :note: ''
-- :commit: 8be1ff11dc1fae61146dbcfaa38e12314d290dca
-  :note: ''
-vccs: []
+- commit: e5787005a9004d7be289cc649c6ae4f3051996cd
+  note: ''
```
Comment (Nonordon) [inline cves/CVE-2016-1647.yml]: Nothing to note for the fixes?

File: cves/CVE-2016-1647.yml [hunk 12]
```diff
@@ -97,20 +104,21 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    No evidence found
```
Comment (Nonordon) [inline cves/CVE-2016-1647.yml]: There is some information here. https://bugs.chromium.org/p/chromium/issues/detail?id=590284

File: cves/CVE-2016-1647.yml [hunk 13]
```diff
@@ -153,38 +161,42 @@ lessons:
...
-    applies: 
-    note: 
+    applies:
+    note:
   security_by_obscurity:
-    applies: 
-    note: 
+    applies: true
+    note: |
+       The vulnerability was largely about how the browser decides, if the user
```
Comment (Nonordon) [inline cves/CVE-2016-1647.yml]: The grammar in this section seems wrong.

File: cves/CVE-2016-1647.yml [hunk 14]
```diff
@@ -32,41 +32,46 @@ description_instructions: |
...
+  note: Doesn't check the condition if navigation within page
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 2
```
Comment (Nonordon) [inline cves/CVE-2016-1647.yml]: I really like this one, 4 upvotes.

File: cves/CVE-2016-1671.yml [hunk 15]
```diff
@@ -46,22 +49,24 @@ reviews:
...
+  note: The person does not check for specific conditions from the outset
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 4
```
Comment (Nonordon) [inline cves/CVE-2016-1671.yml]: I give this 2 upvotes.

File: cves/CVE-2016-1671.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (avj2668) [reply cves/CVE-2016-1671.yml]: Not really, it already written in simple words

File: cves/CVE-2016-1647.yml (hunk 10 above)
Comment (avj2668) [reply cves/CVE-2016-1647.yml]: fixed

**Pull request merged**